consumer/consumererror/partialscrapeerror.go

```diff
@@ -0,0 +1,43 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consumererror
+
+import "errors"
+
+// PartialScrapeError can be used to signalize that a subset of metrics were failed
+// to be scraped
+type PartialScrapeError struct {
```

Comment:
I could be wrong but it feels like this and obsreport stuff about scraping would ideally all be in the same package (or subpackages) instead of spread across core. @bogdandrutu any thoughts?

Reply:
Yea I'm not sure about this one.

I felt the `obsreport` stuff should stay together, and since that package needs to reference `PartialScrapeError` to compute metric stats, I needed to put this error in a separate package as well. `consumererror` seems to be the most appropriate place.

Reply:
let's leave it there for the moment. @james-bebbington you promised me an issue to discuss making Scraper a top level component.

Reply:
Did I? 😅

After applying this library to the hostmetrics receiver, my thoughts were that we could make this library even more usable if we:

- Add `ScraperFactory` interfaces
- Add code to simplify creation of scrapers via factories to `receiver/receiverhelper`

At that point, it may make sense to:

- Move interfaces from `receiver/receiverhelper/scraper.go` into `component/scraper.go`

I don't think that scrapers should be a "top level component" though in the sense that they should be configurable at the top level like receivers / processors / exporters. Here's a doc Jay and I were using a while back to discuss the idea of making Scraper a top level component which eventually led us to do none of the things in that doc and implement this as a library in receiverhelper instead: https://docs.google.com/document/d/15WKpYvw23rQ6WnBhoE2wdk4szfjlXEXg8WYYvBcKshA/edit

Reply:
Please move this to an issue so we don't forget about this

Reply:
https://github.com/open-telemetry/opentelemetry-collector/issues/2036